mkdocs/docs/how-to-release.md

```diff
@@ -143,6 +143,10 @@ https://pypi.org/project/pyiceberg/$VERSION/
 
 And can be installed using: pip3 install pyiceberg==$VERSION
 
+Instructions for verifying a release can be found here:
+
+* https://py.iceberg.apache.org/verify-release/
```

Comment:
I think adding this can help a lot! Especially for those verifying a release for the first time 👍